### ISSUE TITLE
Implementation details with console

### ISSUE THREAD

**Issue by pgulutzan** [tarantool/tarantool#2430]
I have Ubuntu version 14.04.
I have Tarantool version 1.8, pulled from source today.

1. To crash the console, say \set and an SQL statement
without saying box.cfg{}. FOr example:
"
pgulutzan@pgulutzan-TECRA-M10:~/tarantool_sandbox$
~/tarantool-1.8/src/tarantool

version 1.8.0-90-gb91aeb2
type 'help' for interactive help
tarantool> \set language sql
---
- true
...

tarantool> values (1)
builtin/box/load_cfg.lua:322: Please call box.cfg{} first
stack traceback:
 builtin/box/load_cfg.lua:322: in function '__index'
 builtin/box/console.lua:153: in function 'eval'
 builtin/box/console.lua:325: in function 'repl'
 builtin/box/console.lua:373: in function <builtin/box/console.lua:362>
pgulutzan@pgulutzan-TECRA-M10:~/tarantool_sandbox$
"

2. Clarification of the description in issue#2117:
\h[elp]
does not mean you can enter \h or \he or \hel or \help.
You can only enter \h or \help.
I assume this is deliberate, and it's fine.
But for documentation reasons I'd like confirmation
that it is deliberate.

3. I believe it was agreed that upper case would be
allowed, for example
\SET DELIMITER ;
Perhaps there is already an issue report for this,
but I don't see it.

4. Up-arrow (used to repeat previous statement) may not
be working in a way that everyone would expect. For example, say:
box.cfg{}
\set language sql
\set delimiter ;
values (1);
That works fine.
Now I press up-arrow. I see "values (1)". I don't see "values (1);".
Is it deliberate to repeat the statement, but not repeat the delimiter?

5. The delimiter does not apply to so-called meta-commands. For example:
box.cfg{}
\set language sql
\set delimiter ;
\set language lua;
Result = - error: 'Invalid language "lua;", supported languages: lua and
sql.'
Is it deliberate to say delimiter does not apply for what begins with "\"?

6. If I put two SQL statements on one line, the return from the
first statement is ignored. For example (after saying box.cfg):
"
tarantool> \set language sql
---
- true
...

tarantool> \set delimiter ;
---
...

tarantool> VALUES (1); VALUES (2);
---
- - [2]
...
"

**Comment by rtsisyk** [comment]
1.

> 1. To crash the console, say \set and an SQL statement
without saying box.cfg{}. 
> 
> tarantool> values (1)
> builtin/box/load_cfg.lua:322: Please call box.cfg{} first
> stack traceback:
> builtin/box/load_cfg.lua:322: in function '__index'
> builtin/box/console.lua:153: in function 'eval'
> builtin/box/console.lua:325: in function 'repl'
> builtin/box/console.lua:373: in function <builtin/box/console.lua:362>

Fixed in 1.8.0-101-g8c8bdb42d:

```
./src/tarantool
version 1.8.0-100-g38c5400f0
type 'help' for interactive help
tarantool> \set language sql
---
- true
...

tarantool> SELECT * FROM _space;
---
- error: "builtin/box/load_cfg.lua:322: Please call box.cfg{} first\nstack traceback:\n\tbuiltin/box/load_cfg.lua:322:
    in function '__index'\n\tbuiltin/box/console.lua:153: in function <builtin/box/console.lua:153>\n\t[C]:
    in function 'pcall'\n\tbuiltin/box/console.lua:153: in function 'eval'\n\tbuiltin/box/console.lua:325:
    in function 'repl'\n\tbuiltin/box/console.lua:373: in function <builtin/box/console.lua:362>"
...

```

`box.cfg()` can only be called from Lua. Console doesn't call it implicitly.

2. 

> Clarification of the description in issue#2117:
> \h[elp]
> does not mean you can enter \h or \he or \hel or \help.
> You can only enter \h or \help.
> I assume this is deliberate, and it's fine.
> But for documentation reasons I'd like confirmation
> that it is deliberate.

Yes, only `\h` and `\help` forms are supported and that is deliberate.


3.

> I believe it was agreed that upper case would be
> allowed, for example
> \SET DELIMITER ;
> Perhaps there is already an issue report for this,
> but I don't see it.

Both mysql and psql have case-sensitive backslash commands:

```
MariaDB [(none)]> \d ;
MariaDB [(none)]> \D ;
ERROR: Unknown command '\D'.
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '\D' at line 1
MariaDB [(none)]>
```

```
postgres=> \set
AUTOCOMMIT = 'on'
PROMPT1 = '%/%R%# '
PROMPT2 = '%/%R%# '
PROMPT3 = '>> '
VERBOSITY = 'default'
SHOW_CONTEXT = 'errors'
VERSION = 'PostgreSQL 9.6.2 on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-10) 6.3.0 20170321, 64-bit'
DBNAME = 'roman'
USER = 'roman'
HOST = '/var/run/postgresql'
PORT = '5432'
ENCODING = 'UTF8'
postgres=> \SET
Invalid command \SET. Try \? for help.
postgres=>
```

4. 
> Up-arrow (used to repeat previous statement) may not
> be working in a way that everyone would expect. For example, say:
> box.cfg{}
> \set language sql
> \set delimiter ;
> values (1);
> That works fine.
> Now I press up-arrow. I see "values (1)". I don't see "values (1);".
> Is it deliberate to repeat the statement, but not repeat the delimiter?

It works in the same way as Lua console.

5.

> The delimiter does not apply to so-called meta-commands. For example:
> box.cfg{}
> \set language sql
> \set delimiter ;
> \set language lua;
> Result = - error: 'Invalid language "lua;", supported languages: lua and
sql.'
> Is it deliberate to say delimiter does not apply for what begins with ""?

Yes, it is deliberate. Meta-commands ain't affected by delimiter.

6.
>     f I put two SQL statements on one line, the return from the
>    first statement is ignored. For example (after saying box.cfg):
>    "
>    tarantool> \set language sql
>
>    true
>    ...
> tarantool> \set delimiter ;
> ...
> tarantool> VALUES (1); VALUES (2);

It is not a problem of console:

```
tarantool> box.sql.execute("VALUES(1); VALUES(2)")
---
- - [2]
...
```

We plan to discuss this problem, I'll post an update later.

**Comment by kostja** [comment]
The only remaining problem of this issue was moved out to gh-2559
